test/delta_method.jl

```diff
@@ -37,55 +38,99 @@ using Test
     @test isapprox(only(eff_emm_trans.err), 1.07; atol=0.005)
     @test isapprox(only(eff_emm_trans.lower), 45.3; atol=0.05)
     @test isapprox(only(eff_emm_trans.upper), 47.5; atol=0.05)
+
+    @testset "AutoInvLink fails gracefully" begin
+        # this should work even pre Julia 1.9 because by definition
+        # no extension is loaded
+        @test_throws ArgumentError effects(design, model; invlink=AutoInvLink())
+    end
 end
 
 @testset "link function" begin
     dat = rdataset("car", "Cowles")
     dat[!, :vol] = dat.Volunteer .== "yes"
     model = glm(@formula(vol ~ Extraversion * Neuroticism), dat, Bernoulli())
-    invlink = Base.Fix1(GLM.linkinv, Link(model.model))
     design = Dict(:Extraversion => [13],
                   :Neuroticism => [16])
     X = [1.0 13.0 16.0 13 * 16]
+    iv = Base.Fix1(GLM.linkinv, Link(model.model))
+    @static if VERSION >= v"1.9"
+        invlinks = [iv, AutoInvLink()]
+        @test Effects._model_link(model, AutoInvLink()) ==
+              Effects._model_link(model.model, AutoInvLink())
+    else
+        invlinks = [iv]
+    end
+    @testset "invlink = $invlink" for invlink in invlinks
+        for level in [0.68, 0.95]
+            eff = effects(design, model; invlink, level)
+
+            # compare with results from GLM.predict
+            pred = DataFrame(predict(model.model, X;
+                                     interval=:confidence,
+                                     interval_method=:delta,
+                                     level))
+            @test all(pred.prediction .≈ eff.vol)
+            @test all(isapprox.(pred.lower, eff.lower; atol=0.001))
+            @test all(isapprox.(pred.upper, eff.upper; atol=0.001))
 
-    for level in [0.68, 0.95]
-        eff = effects(design, model; invlink, level)
+            eff_trans = effects(design, model; level)
+            transform!(eff_trans,
+                       :vol => ByRow(iv),
+                       :lower => ByRow(iv),
+                       :upper => ByRow(iv); renamecols=false)
+            # for this model, things play out nicely
+            @test all(eff_trans.vol .≈ eff.vol)
+            @test all(isapprox.(eff_trans.lower, eff.lower; atol=0.001))
+            @test all(isapprox.(eff_trans.upper, eff.upper; atol=0.001))
```

Comment:
```suggestion
            @test eff_trans.vol ≈ eff.vol
            @test eff_trans.lower ≈ eff.lower atol=0.001
            @test eff_trans.upper ≈ eff.upper atol=0.001
```
Same question as above

Reply:
same answer as above